src/shims/env.rs

```diff
@@ -55,7 +55,7 @@ impl<'tcx> EnvVars<'tcx> {
                 };
                 if forward {
                     let var_ptr = match target_os {
-                        "linux" | "macos" =>
+                        "linux" | "macos" | "freebsd" =>
```

Comment:
As mentioned in the Android PR, this check should be moved into a separate function. Please make a separate PR that does only that.

Reply:
well I noticed that the android PR had the change already, so I didn't just want to yoink the diff from the android PR
but if you insist I can take their changes and make it into its own PR